app/src/main/java/co/nimblehq/compose/crypto/ui/screens/detail/DetailViewModel.kt

```diff
@@ -32,19 +45,67 @@ class DetailViewModel @Inject constructor(
     override val coinDetailUiModel: StateFlow<CoinDetailUiModel?>
         get() = _coinDetailUiModel
 
+    private val _coinPrices = MutableStateFlow<List<CoinPrice>>(emptyList())
+    override val coinPrices: StateFlow<List<CoinPrice>>
+        get() = _coinPrices
+
     override fun getCoinId(coinId: String) {
         getCoinDetail(coinId = coinId)
+        getCoinPrices(coinId = coinId)
     }
 
     private fun getCoinDetail(coinId: String) = execute {
         showLoading()
         getCoinDetailUseCase.execute(coinId = coinId)
             .catch { e ->
+                hideLoading()
                 _error.emit(e)
             }
             .collect { coinDetail ->
+                hideLoading()
                 _coinDetailUiModel.emit(coinDetail.toUiModel())
             }
         hideLoading()
```

Comment:
we have `hideLoading()` at line 68 in case of error already

```suggestion
        showLoading()
        getCoinDetailUseCase.execute(coinId = coinId)
            .catch { e ->
                _error.emit(e)
            }
            .collect { coinDetail ->
                _coinDetailUiModel.emit(coinDetail.toUiModel())
            }
        hideLoading()
```